### ISSUE TITLE
HierarchicalSampler is a `batch_sampler` but should extend `torch.utils.data.Sampler`, not `torch.utils.data.BatchSampler`

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#613]

**Comment by KevinMusgrave** [comment]
Fixed in v2.1.2

```
pip install pytorch-metric-learning==2.1.2
```
